src/main/scala/za/co/absa/commons/lang/converters.scala

```diff
@@ -34,10 +34,16 @@ trait Converter {
   * Memoized converter
   */
 trait CachingConverter extends Converter {
-  private val cache = mutable.LinkedHashMap.empty[From, To]
+  private val cache = mutable.LinkedHashMap.empty[Key, To]
 
   def values: Seq[To] = cache.values.toSeq
 
+  protected type Key = Any
+  protected def key(x: From): Key = x
+
   abstract override def convert(arg: From): To =
-    cache.getOrElseUpdate(arg, super.convert(arg))
+    cache.getOrElseUpdate(
+      key(arg),
+      super.convert(arg)
```

Comment:
I just wonder if this does, what I believe you want to do.
I think what you want to do with this trait, is avoid multiple conversion if the conversion happened before, right?
But I think the conversion will still happen, as it's evaluated to be passed as parameter to `getOrElseUpdate`,, isn't it? 🤔

Reply:
> I think what you want to do with this trait, is avoid multiple conversion if the conversion happened before, right?

Yes.

> But I think the conversion will still happen, as it's evaluated to be passed as parameter to getOrElseUpdate,, isn't it?

No. The second parameter of the `getOrElseUpdate()` method is called by-name, meaning lazily.

Reply:
Ahh, yeah. 👍